networkx/algorithms/shortest_paths/tests/test_dense.py

```diff
@@ -193,20 +201,62 @@ def test_zero_distance(self):
         # make sure we get lower weight
         # to_undirected might choose either edge with weight 2 or weight 3
         GG["u"]["x"]["weight"] = 2
-        path, dist = nx.floyd_warshall_predecessor_and_distance(GG)
+        path, dist = floyd_fn(GG)
 
         for u in GG:
-            dist[u][u] = 0
+            assert dist[u][u] == 0
 
-    def test_zero_weight(self):
+    def test_zero_weight(self, floyd_fn):
         G = nx.DiGraph()
         edges = [(1, 2, -2), (2, 3, -4), (1, 5, 1), (5, 4, 0), (4, 3, -5), (2, 5, -7)]
         G.add_weighted_edges_from(edges)
         dist = nx.floyd_warshall(G)
         assert dist[1][3] == -14
```

Comment:
Should these two lines (testing floyd_warshall(G)) be removed when you add the `floyd_fn` tests below it?

Similarly 4 lines below